config.js

```diff
@@ -2,6 +2,35 @@
 
 const path = require('path');
 
+let environment = process.env;
+
 module.exports = {
-    BIN_OUT_PATH: path.join(__dirname, 'lib', 'chromium')
+    /**
+     * Set an alternative "env" object. If not set will use process.env.
+     * This is intended for mocking out the env in unit tests.
+     * @param env Object exposing key/value pairs of environment variables. Pass a falsey value to unset.
+     */
+    _setEnv: env => {
+        environment = env || process.env;
+    },
+    BIN_OUT_PATH: path.join(__dirname, 'lib', 'chromium'),
+    /**
+     * Gets a configuration parameter from the environment.
+     * Will first check for a lowercase variant set via npm config in the format: `npm_config_${name.toLowerCase()}`.
+     * If not set then will check the environment for the variable, as provided.
+     * @param {string} name The name of the environment variable, case sensitive, conventionally uppercase.
+     * @returns {string} The value of the environment variable.
+     */
+    getEnvVar: name => {
```

Comment:
I got the only question for this new functionality.
I'm absolutely fine with using `.npmrc` file variable but I wouldn't like to create some specific naming conventions because of this (lower-case).
There are a few points here:
1. If we are talking about env variables, it's used to have them in upper-case, that's why I would prefer `.npmrc` to have then upper-case as well (e.g. `CHROMIUM_DOWNLOAD_HOST=...`)
2. I'm not very fond of having `_setEnv()` function that is needed **only** for testing purpose inside the "prod" module. I do understand that it's here because of the ease of use with `getEnvVar()` function but still I don't like it being here. Can't we just put variables that we need in tests using: `process.env['var'] = 'value';` instead of creating separate env?
3. Can't we simplify this function to something like: 
```
getEnvVar: name => return process.env[`npm_config_${name}`] || process.env[name] || '';
```

What do you think about this, Rick?

Reply:
Hey Dmytro, good questions :)

1. This Env.UPPERCASE and npm_config_lowercase is pretty much convention in the NodeJS world, not something I or psociety came up with. For example, from [the Puppeteer docs](https://github.com/puppeteer/puppeteer/blob/v5.2.1/docs/api.md#environment-variables):

>Puppeteer looks for certain environment variables to aid its operations. If Puppeteer doesn't find them in the environment during the installation step, a lowercased variant of these variables will be used from the npm config.

Or from this [fork of node-chromium](https://github.com/giggio/node-chromedriver/blob/6743dda246ab1e7262e57fb89275a79ec838a5a3/install.js#L18):

> let cdnUrl = process.env.npm_config_chromedriver_cdnurl || process.env.CHROMEDRIVER_CDNURL 

Huh, interesting I could have just used that fork! But we've come this far ;)

So I'd say, stick to convention, UPPERCASE for environment variables, lowercase for npmrc. Ultimately it's your codebase and your final decision...

2. I totally agree, I don't like it either. JS simply doesn't have enough modifiers, it needs `protected` or `package` or `friend`. Since it doesn't have these the convention is to name such methods with a leading underscore, hence `_getEnv`. This signals that it is not really part of your API, it is pseudo private, anyone who uses it does so at their own risk. 

Personally I think testing is a first class customer of your code, and it is worth putting affordances on your API to facilitate it.

I'm happy to get rid of it and revert to `process.env['var'] = 'value';`. No big deal. The worst I See happening is that your tests will end up having to know a whole long list of environment variables to reset. I predict node-chromium will allow an increasing number of configuration properties via the environment, just like [this fork](https://github.com/giggio/node-chromedriver).

Then, in the tests, there would need to be a big reset (it could be a util) something like this:

```
test.beforeEach(t => {
    process.env.CHROMIUM_REVISION = '';
    process.env.npm_config_chromium_revision = '';
    process.env.CHROMIUM_DOWNLOAD_HOST = '';
    process.env.npm_config_chromium_download_host = '';
   // AND so on for every environment variable added in the future
}
```

3. Let me think about that tomorrow - it's getting late here :) Not sure if I'm even making sense anymore.

Reply:
Thank you for your reply, Rick! I do agree with 1.
Regarding 2., I prefer tests to make theirs infrastructure themselves. It can be an util function inside tests itself that will: create "empty" env, dump existing and override `proccess.env` with the new "empty" env. After the test the real env can be restored from the dump.
The problem I see with "setting" env as you do it now is that person should always remember to unset env after the test. And, even more problem arise if tests depend/will depend on some env variables available in system (e.g. PATH, OS, ...).

Thank you for your help and the link to the fork. I, actually, has never seen it and it looks very interesting.

Reply:
Just took a look into the fork and it's not actually a fork but the similar package for ChromeDriver and not Chromium. But still, very interesting. Thank you.

Reply:
2. OK it seems quite easy to override `process.env` in NodeJS - at least in the version I am running. You can simply blat it `process.env = {}` so I can easily get rid of the ugly `_setEnv` method, it's a good thing you questioned it 👍 

BTW we don't need to bother restoring `process.env` - we can completely stomp all over it and leave it messed up because of [AVA spawns a fresh NodeJS process for each test file](https://github.com/avajs/ava#why-not-mocha-tape-tap):

>   AVA is highly opinionated and runs tests concurrently, with a separate process for each test file.

3. Yes it can be much simpler than I wrote it.

Reply:
You are absolutely right. Nice!